From 081bc869311fed6b868f77576b90da0b01cae3ce Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 11 Jun 2018 17:45:29 +0200
Subject: [PATCH 0802/1200] arm64: allow to use platform specific sync abort

Thanks to this weak handler, any platform will be able to easily
register it's own sync abort handler.

Change-Id: I3a2c653def070d1d49126cd7a9db290620184f36
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/56765
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57738
---
 arch/arm/lib/interrupts_64.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/lib/interrupts_64.c b/arch/arm/lib/interrupts_64.c
index 2e1a7a45dd..cf263b3363 100644
--- a/arch/arm/lib/interrupts_64.c
+++ b/arch/arm/lib/interrupts_64.c
@@ -91,6 +91,9 @@ void do_bad_error(struct pt_regs *pt_regs, unsigned int esr)
 	panic("Resetting CPU ...\n");
 }
 
+__weak void plat_do_sync(void)
+{
+}
 /*
  * do_sync handles the Synchronous Abort exception.
  */
@@ -98,6 +101,8 @@ void do_sync(struct pt_regs *pt_regs, unsigned int esr)
 {
 	efi_restore_gd();
 	printf("\"Synchronous Abort\" handler, esr 0x%08x\n", esr);
+
+	plat_do_sync();
 	show_regs(pt_regs);
 	panic("Resetting CPU ...\n");
 }
-- 
2.22.0

